engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/NameRecordingScreen.java

```diff
@@ -0,0 +1,202 @@
+     *
+     * @param newTitle The title of the new recording.
+     */
+    private void loadGame(String newTitle) {
+        try {
+            final GameManifest manifest = gameInfo.getManifest();
+
+            copySaveDirectoryToRecordingLibrary(manifest.getTitle(), newTitle);
+            recordAndReplayUtils.setGameTitle(newTitle);
+            config.getWorldGeneration().setDefaultSeed(manifest.getSeed());
+            config.getWorldGeneration().setWorldTitle(newTitle);
+            CoreRegistry.get(GameEngine.class).changeState(new StateLoading(manifest, NetworkMode.NONE));
+        } catch (Exception e) {
+            logger.error("Failed to load saved game", e);
+            getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Error Loading Game", e.getMessage());
+        }
+    }
+
+    /**
+     * Copies the selected save files to a new recording directory.
+     *
+     * @param oldTitle The name of the original save directory.
+     * @param newTitle The name of the new recording directory.
+     */
+    private void copySaveDirectoryToRecordingLibrary(String oldTitle, String newTitle) {
+        File saveDirectory = new File(PathManager.getInstance().getSavePath(oldTitle).toString());
+        Path destinationPath = PathManager.getInstance().getRecordingPath(newTitle);
+        File destDirectory = new File(destinationPath.toString());
+        try {
+            FileUtils.copyDirectoryStructure(saveDirectory, destDirectory);
+            rewriteManifestTitle(destinationPath, newTitle);
+        } catch (Exception e) {
+            logger.error("Error trying to copy the save directory:", e);
+        }
+    }
+
+    /**
+     * Rewrites the title of the save game manifest to match the new directory title.
+     *
+     * @param destinationPath The path of the new recording files.
+     * @param newTitle The new name for the recording manifest.
+     * @throws IOException
+     */
+    private void rewriteManifestTitle(Path destinationPath, String newTitle) throws IOException {
+        // simply grabs the manifest, changes it, and saves again.
+        GameManifest manifest = GameManifest.load(destinationPath.resolve(GameManifest.DEFAULT_FILE_NAME));
+        manifest.setTitle(newTitle);
+        GameManifest.save(destinationPath.resolve(GameManifest.DEFAULT_FILE_NAME), manifest);
+    }
+
+    /**
+     * Tests if the provided string is valid for a game name.
+     *
+     * @param name The provided name string.
+     * @return true if name is valid, false otherwise.
+     */
+    private boolean isNameValid(String name) {
+        Path destinationPath = PathManager.getInstance().getRecordingPath(name);
+
+        // invalid characters are filtered, so if the file name is made up of entirely invalid characters, the path will have a blank name.
+        if (destinationPath == PathManager.getInstance().getRecordingPath("")) {
```

Comment:
The `==` is a reference comparison, here you are checking if both objects point to the same memory location, and of course, they don't. The right way to compare these paths is using `equals()`. As shown below, for example, if I put "@" as the recording name, which does not match the `PathManager` regex (`[^A-Za-z0-9-_ ]`), the `isNameValid(name)` method returns `true`, again because you're not comparing the object value itself.

Moreover, it can cause other bugs like the one shown in the pic, saying that a recording already exists.

![image](https://user-images.githubusercontent.com/26657147/48665079-4130c580-ea87-11e8-93be-8e58d6dd4227.png)

Reply:
Thanks for the great explanation, I saw that there was an `equals()` function, but I didn't know that there was such a big difference from `==`, I guess I learned something new today :)

The `==` comparison has been changed to an 'equals()' comparison, and I cleaned up the function a bit afterwards by removing the blank string check at the end, since blank input is already handled by the first comparison. Now, entering "@" no longer results in the wrong error message.

Reply:
You're welcome! As you noticed the the wrong error message was a consequence of the wrong comparison. Since `isNameValid(name)` was returning `true` when it should be `false`, the second `if` statement on `enterPressed()` was executed causing this bug.